packages/jest-diff/src/__tests__/diff-test.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @emails oncall+jsinfra
+ */
+
+'use strict';
+
+jest.disableAutomock();
+
+const diff = require('../');
+const stripAnsi = require('strip-ansi');
+
+describe('different types', () => {
+  [
+    [1, 'a', 'number', 'string'],
+    [{}, 'a', 'object', 'string'],
+    [[], 2, 'array', 'number'],
+    [null, undefined, 'null', 'undefined'],
+    [() => {}, 3, 'function', 'number'],
+  ].forEach(values => {
+    const a = values[0];
+    const b = values[1];
+    const typeA = values[2];
+    const typeB = values[3];
+
+    test(`'${a}' and '${b}'`, () => {
+      expect(stripAnsi(diff(a, b))).toBe(
```

Comment:
this actually would be super awesome, because it can easily work on all the colors and stuff that's close to impossible to test manually.
The only thing is that we'll need to use babel-node or something, because right now the snapshots will be saved to `build/__tests__/__snapshots__`